src/dotnet/commands/dotnet-vstest/VSTestForwardingApp.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Cli.Utils;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class VSTestForwardingApp
+    {
+        private const string vstestExename = "vstest.console.dll";
+        private readonly ForwardingApp _forwardingApp;
+
+        public VSTestForwardingApp(IEnumerable<string> argsToForward)
+        {
+            _forwardingApp = new ForwardingApp(
+                GetVSTestExePath(),
+                argsToForward,
+                environmentVariables: GetEnvironmentVariables()
+                );
+        }
+
+        public int Execute()
+        {
+            return _forwardingApp.Execute();
+        }
+
+        private Dictionary<string, string> GetEnvironmentVariables()
+        {
+            return new Dictionary<string, string>
+            {
+                { "DotnetHostPath", GetHostPath() },
```

Comment:
Is this environment variable respected by vstest.console?  Would it be better if this was passed as a command line arg instead of an Environment Variable?